### DISCUSSION TITLE
Welcome to Mammotion-HA Discussions!

### DISCUSSION THREAD

**Discussion by mikey0000** [mikey0000/Mammotion-HA#55] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by zipposi** [comment]
Hello!

I'm a medium-grade Home Assistant (HA) user and I would kill to have a Luba integration that offers me the same properties that the application has, but:

- Is not clumsy as the app is (I really hate it...).
- Has access to areas, tasks or let me compose a task from HA.
- Gives control of the Luba while is working (WiFi needed).

I have millions of ideas and just waiting to get a functional integration to do this examples below. Of course I mean to do this with the help of other HA sensors (rain, workday,.. etc.).

- Begin working when there is nobody home.
- Do not even start if is raining.
- Do not even start if not a workday.
- Cut higher if is summer or lower if is one of the other seasons.
- Cut in the night if is too hot.

...etc. I can go wild if I get a good control over Luba !:)

So just wanted to let you know that you hav an avid fan here, just waiting for the WiFi option, because the Bluetooth is not a viable way for my case. ;)

Kind regards,
m@rko

**Reply by mikey0000** [reply]
All these will be doable in good time 😁. A lot of us are hanging out in the discord server if you'd like to keep updated with progress and so on.

**Comment by Jogi83** [comment]
I'm really looking forward to the integration too, and everything that it will make possible. Maybe even the following:
My Luba has to mow a part of my driveway and I'm afraid that one day a parcel service will run over my Luba while reversing. It would be great if I could tell my Luba to drive to another zone (or briefly drive to a waiting position) when the surveillance camera detects a car driving onto my property.

**Comment by PainInc1** [comment]
Thank you! Thank you for the latest update (0.2.7) made to the integration! Finally my Luba 3KX starts mowing triggered by my automation. What ever you've done...after 1,5 month non doing anything he starts its job. Well done! Keep going and with the best regards from Germany!

**Reply by mikey0000** [reply]
Hey, no worries another user reported an issue and I was able to figure it out with the use of their bot

**Comment by Pascal76B** [comment]
Hello, I'm developing my own personal home automation system, and I'd like to integrate my Mammotion Yuka Mini robot. I managed to connect it via the cloud, but the updates aren't real-time. I came across your script, but I can't figure out how it works. I'd like to find the access details to connect to my robot live through my Mammotion account. Do you have any help with this? How do I find the commands to connect? I hope you'll take the time to reply and help me.

Sincerely, Pascal


bonjour , je suis en train de dev mon propre HA perso , j'aimerais y integré mon robot mammotion Yuka mini , j'ai reussi a passer par le CLOUD et le connecter mais les mise à jour sont pas en temps reel , je suis tombé sur votre script , mais impossible de comprendre comment il marche , j'aimerais y trouver les acces pour me connecter en live a mon robot via mon compte mammotion , avez une aide pour cela ? comment trouver les commandes pour se connecter . esperant que vous prendrez le temps de me repondre pour m'aider .

Amicalement Pascal

**Reply by mikey0000** [reply]
It is real time, you need to request updates, essentially poll. Look at the home assistant api files in pymammotion

**Reply by Pascal76B** [reply]
Unfortunately, I couldn't find any information on how to connect live; only the cloud works, and it's not live.

malheureusement j'ai pas trouvé d'infos pour me connecter en live , seul le cloud marche est c'est pas en live

**Reply by mikey0000** [reply]
Ble and cloud both work and they are both live. I'm not sure why your arguing with me over this, I'm telling you it is live. If you send a command you will get an instant response over cloud or ble unless you've missed entirely the MQTT connection.

**Reply by mikey0000** [reply]
The best way for you to learn how to use pymammotion is from the home assistant init and coordinator files and the test files in pymammotion
